src/util/file_log.rs

```diff
@@ -120,13 +117,21 @@ impl RotatingFileLogger {
     }
 }
 
-impl LogWriter for RotatingFileLogger {
-    fn write(&self, args: Arguments) {
+impl Write for RotatingFileLogger {
+    fn write(&mut self, bytes: &[u8]) -> Result<usize, ::std::io::Error> {
+        let mut core = self.core.lock().unwrap();
+        if core.should_rollover() {
+            core.do_rollover()
+        };
+        core.file.write(bytes)
+    }
+
+    fn flush(&mut self) -> Result<(), ::std::io::Error> {
         let mut core = self.core.lock().unwrap();
         if core.should_rollover() {
             core.do_rollover()
         };
-        let _ = core.file.write_fmt(args);
+        core.file.flush()
```

Comment:
Should it be flushed first before rollover?

Reply:
That makes sense to me!